gordon/interfaces.py

```diff
@@ -120,3 +120,41 @@ class IPublisherClient(IGenericPlugin):
         Args:
             event_msg (IEventMessage): Message ready to be sent to destination.
         """
+
+
+class IMetricRelay(Interface):
+    """Create and publish (push) metrics on demand."""
+
+    def incr(metric_name, value=1):
+        """Increment a metric. Create one if new."""
+
+    def timer(metric_name):
+        """Return a :class:`ITimerMetric`. Create one if new."""
+
+    def set_counter(metric_name, counter):
+        """Set a :class:`ICounterMetric` object to a given metric name."""
+
+    def set_timer(metric_name, timer):
+        """Set a :class:`ITimerMetric` to a given metric name."""
+
+
+class IGenericMetric(Interface):
+    """**Do not** implement this interface directly.
+
+    Use :class:`ICounterMetric` or :class:`ITimerMetric`
+    instead.
+    """
+
+
+class ICounterMetric(IGenericMetric):
```

Comment:
Do you think we will need a gauge-type metric as well for reporting changing numbers? For example, Gordon may jump between reporting a high number of messages (if the janitor sends out a lot of changes) and a low number if things are particularly quiet.

I'm not sure if a Counter metric would satisfy this situation in a long-running service like Gordon. I think that in cases like this we used to use shumway's Metric objects to fire off one off metrics.

Reply:
This is outdated convo, right? There is a Gauge below.